### ISSUE TITLE
CosmoDB - using serverless with gremlin results in an error when creating the database

### ISSUE THREAD

**Issue by lawrenceong** [terraform-providers/terraform-provider-azurerm#11062]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```sh
Terraform v0.14.8
+ provider registry.terraform.io/hashicorp/azurerm v2.52.0
```

### Affected Resource(s)

* `azurerm_cosmosdb_gremlin_database`

### Terraform Configuration Files

Create an Azure CosmoDB with Gremlin capability using Serverless mode as such:

```hcl
resource "azurerm_cosmosdb_account" "db" {
  name                = local.db_name
  location            = data.azurerm_resource_group.rg.location
  resource_group_name = data.azurerm_resource_group.rg.name
  offer_type          = "Standard"
  kind                = "GlobalDocumentDB"

  capabilities {
    name = "EnableGremlin"
  }

  capabilities {
    name = "EnableServerless"
  }

  consistency_policy {
    consistency_level = "Session"
  }

  geo_location {
    location          = data.azurerm_resource_group.rg.location
    failover_priority = 0
  }
}
```

Attempt to create a Gremlin database:

```hcl
resource "azurerm_cosmosdb_gremlin_database" "graphdb" {
  name                = "graphdb"
  resource_group_name = azurerm_cosmosdb_account.db.resource_group_name
  account_name        = azurerm_cosmosdb_account.db.name
}
```

### Debug Output

https://gist.github.com/lawrenceong/4b36eb51fc20fede93a8dc6d4c437e1f

### Expected Behaviour

No errors with DB creation.

### Actual Behaviour

The following error appeared post creation of the database:

> Error: Error reading Throughput on Cosmos Gremlin Database "graphdb" (Account: "test-1281"): documentdb.GremlinResourcesClient#GetGremlinDatabaseThroughput: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="Message: {\"code\":\"BadRequest\",\"message\":\"Reading or replacing offers is not supported for serverless accounts.\\r\\nActivityId: 57b9c6ea-f108-4257-a26a-01a3d6825070, Microsoft.Azure.Documents.Common/2.11.0\"}, Request URI: /offers, RequestStats: , SDK: Microsoft.Azure.Documents.Common/2.11.0"

### Steps to Reproduce

1. `terraform apply`

**Comment by sMuro** [comment]
Got this same error, and after that I get error in plan phase

>Error: Error waiting on create/update future for Cosmos Gremlin Database "graphdb" (Account: "gremlindb"): Code="BadRequest" Message="Message: {\"code\":\"BadRequest\",\"message\":\"Shared throughput database creation is not supported for serverless accounts.\\r\\nActivityId: bc0a8b6b-0fc4-41a5-8e19-abe4ce7769e4, Microsoft.Azure.Documents.Common/2.11.0\"}, Request URI: /dbs, RequestStats: , SDK: Microsoft.Azure.Documents.Common/2.11.0, Microsoft.Azure.Documents.Common/2.11.0, Microsoft.Azure.Documents.Common/2.11.0, Microsoft.Azure.Documents.Common/2.11.0, Microsoft.Azure.Documents.Common/2.11.0, Microsoft.Azure.Documents.Common/2.11.0, Microsoft.Azure.Documents.Common/2.11.0, Microsoft.Azure.Documents.Common/2.11.0, Microsoft.Azure.Documents.Common/2.11.0"

**Comment by desertfoxmb** [comment]
Seeing the same issue for both SQL and Gremlin API database account creation with provider 2.78

**Comment by desertfoxmb** [comment]
Any update?

**Comment by rcskosir** [comment]
Thanks for opening this issue. This was a problem in the 2.x version of the provider which is no longer actively maintained.
If this is still an issue with the 4.x version of the provider please do let us know by opening a new issue, thanks!
